src/Sql/Sql.Test/ScenarioTests/Common.ps1

```diff
@@ -738,6 +738,56 @@ function Create-ManagedInstanceForTest ($resourceGroup, $subnetId)
 	return $managedInstance
 }
 
+<#
+	.SYNOPSIS
+	Creates a managed instance in an instance pool
+#>
+function Create-ManagedInstanceInInstancePoolForTest ($instancePool)
+{
+    $managedInstanceName = Get-ManagedInstanceName
+    $credentials = Get-ServerCredential
+    $vCore = 2
+    $managedInstance = $instancePool | New-AzSqlInstance -Name $managedInstanceName -VCore $vCore -AdministratorCredential $credentials -StorageSizeInGb 32 -PublicDataEndpointEnabled
+    return $managedInstance
+}
+
+function Remove-ManagedInstancesInInstancePool($instancePool)
+{
+    $instancePool | Get-AzSqlInstance | Remove-AzSqlInstance -Force
+}
+
+
+function Get-InstancePoolTestProperties()
+{
+    $tags = @{ instance="Pools" };
+    $instancePoolTestProperties = @{
+        resourceGroup = "instancePoolCSSdemo"
+        name = "cssinstancepool0"
+        subnetId = "/subscriptions/2e7fe4bd-90c7-454e-8bb6-dc44649f27b2/resourceGroups/instancePoolCSSdemo/providers/Microsoft.Network/virtualNetworks/vnet-cssinstancepool0/subnets/InstancePool"
```

Comment:
@johnpaulkee we should avoid hard-coding resource ids when we can -- with this strings, it may not be possible for other users to re-record this test if needed since they may not have access to the given subscription and/or resource(s)

Reply:
@cormacpayne for this case, Instance pool creation requires a preconfigured subnet similar to managed instance.

If I gave some instructions here on how to pre-configure their subnet, is this enough?
The ability for us to re-use the subnet would be great.
We do something similar in this method - CreateAndGetVirtualNetworkForManagedInstance

Reply:
Why aren't you just calling CreateAndGetVirtualNetworkForManagedInstance here?

Reply:
We can update to use this.
We can reuse subnet if existing with developer's subscription, but will create if not existing.

Reply:
If there isn't a way to do the creation of the resource(s) during test setup, then providing details as a comment in the script on how to construct the necessary resource(s) is sufficient. We just need to make sure that anyone can run the test in record mode, or has the proper instructions on how to do so.